src/distilabel/tasks/critique/ultracm.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+from dataclasses import dataclass
+from typing import ClassVar
+
+from typing_extensions import TypedDict
+
+from distilabel.tasks.base import get_template
+from distilabel.tasks.critique.base import CritiqueTask
+
+_ULTRACM_TEMPLATE = get_template("ultracm.jinja2")
+
+
+class UltraCMOutput(TypedDict):
+    """A `TypedDict` matching the output format of UltraCM."""
+
+    score: float
+    critique: str
+
+
+@dataclass
+class UltraCMTask(CritiqueTask):
+    __jinja2_template__: ClassVar[str] = _ULTRACM_TEMPLATE
+
+    system_prompt: str = (
+        "User: A one-turn chat between a curious user and an artificial intelligence"
+        " assistant. The assistant gives helpful, very detailed, and polite answers to"
+        " the user's questions.</s>"
+    )
+
+    def generate_prompt(self, instruction: str, completion: str) -> str:
+        render_kwargs = {
+            "instruction": instruction,
+            "completion": completion,
+        }
+        return f"{self.system_prompt}\nUser: {self.template.render(**render_kwargs)}</s>\nAssistant: ### Feedback\nOverall Score: "
```

Comment:
We should return a `Prompt` here, right? We can add an `if` condition in the `Prompt.format_as` for `UltraCM`. I guess this class is meant to be used only with the `UltraCM` model, but I would prefer to not add this specificity and follow the `Task` interface.

Reply:
Indeed for e.g. Llama2 we're doing the same, as that's constrained to a model, and the UltraCM format is not a format per se, so I would keep it this way for the moment, as the `LLM` is ready to support both `str` and `Prompt`, and returning a `Prompt` here just adds complexity for something that's model specific and won't be used in any other scenario outside this one :/